### ISSUE TITLE
Unbundle recyclerview support from main module

### ISSUE THREAD

**Issue by plastiv** [umano/AndroidSlidingUpPanel#563]
Is it possible to make recyclerview sliding panel as separate module, since it requires addition support-recyclerview dependency which is not necessary for projects which are not using it and suffer from 65k problem?

**Comment by plastiv** [comment]
There are couple of ways to do it:

* Create a separate module which depends on the base module and the optional dependency
* Create a module per optional dependency which the base module depends on.
* Mark the dependency as non-transitive and activate the functionality at runtime if present
* Use reflection and activate the functionality at runtime if present
* Use a Service Provider Interface and call the correct implementation at runtime

Runtime check implementation could be reviewed here: https://github.com/umano/AndroidSlidingUpPanel/pull/572

**Comment by nzakharchenko** [comment]
Yes, I have the same problem with 65k on my project, and do not use recyclerview, so would be really cool if you could remove it. Many thanks in advance!

**Comment by izmajlowiczl** [comment]
+1
